### PR TITLE
Closes #5444: IntentReceiverActivity: Add tests for intent processing

### LINKED ISSUE #5444: Make IntentReceiverActivity more testable & write test for intent processing
@csadilek and I spent a considerable amount of time today trying to add additional tests for https://github.com/mozilla-mobile/fenix/pull/5416 but were unable to do to the fact that `IntentReceiverActivity` does not launch intents properly and uses the activity context.

We should make this file more testable so we can add a test for intent processing.

### PR DESCRIPTION
**Pull request by csadilek**
This just covers the intent processing parts mentioned in #5444 and where we were stuck on yesterday for https://github.com/mozilla-mobile/fenix/pull/5416.

### CODE REVIEW

**Comment by csadilek** [comment]
Adding "do not land". Still working on a fix.

**Comment by csadilek** [comment]
ok, this is ready now.

File: app/src/test/java/org/mozilla/fenix/IntentReceiverActivityTest.kt [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    fun `process intent with alwaysOpenInPrivateMode set to true`() {
+        runBlockingTest {
+            testContext.settings.alwaysOpenInPrivateMode = true
+
+            val intent = Intent()
+            `when`(testContext.components.intentProcessors.privateIntentProcessor.process(intent)).thenReturn(true)
+            val activity = Robolectric.buildActivity(IntentReceiverActivity::class.java, intent).get()
+            activity.processIntent(intent)
+
+            // Not using mockk here because process is as suspend function
```
Comment (sblatz) [inline app/src/test/java/org/mozilla/fenix/IntentReceiverActivityTest.kt]: nit typo on all of these comments: "is as suspend function" -> "is a suspend function"

Comment (csadilek) [reply app/src/test/java/org/mozilla/fenix/IntentReceiverActivityTest.kt]: Ah, thanks for fixing that!

**Pull request merged**